Upgrade versions of Eureka Client

Tetragramato opened the pull request:
- Upgrade version for Eureka Archaius in 0.7.4.
- Upgrade Eureka Client in 1.4.12.

This allow to register legacy applications in a more upgraded version of Eureka Server, and avoid Heartbeat error when registering, like :

`
ERROR c.n.d.p.DiscoveryJerseyProvider - Unexpected error occurred during de-serialization of discovery data, done connection cleanup
ERROR c.netflix.discovery.DiscoveryClient - DiscoveryClient_XXXXX-APP0 - was unable to refresh its cache! status = null
javax.ws.rs.WebApplicationException: null
at com.netflix.discovery.provider.DiscoveryJerseyProvider.readFrom(DiscoveryJerseyProvider.java:110)
at com.sun.jersey.api.client.ClientResponse.getEntity(ClientResponse.java:634)
at com.sun.jersey.api.client.ClientResponse.getEntity(ClientResponse.java:586)
at com.netflix.discovery.DiscoveryClient.getAndStoreFullRegistry(DiscoveryClient.java:1126)
at com.netflix.discovery.DiscoveryClient.fetchRegistry(DiscoveryClient.java:1029)
at com.netflix.discovery.DiscoveryClient.access$2100(DiscoveryClient.java:123)
`

Commit:
Upgrade versions of Eureka Client

- Upgrade version for Eureka Archaius in 0.7.4.
- Upgrade Eureka Client in 1.4.12.

sawano commented:
Thank you for the PR!

...and sorry for the slow response.

Have you tried out the changes in a "real" setup so you know it works? (There are currently no "integration" type tests that will verify that automatically so it needs to be done manually)

Also, could you please remove the version bump in the pom, i.e. leave it at 0.0.6-SNAPSHOT. The version is stepped up when a new release is made.

Tetragramato commented:
Hi,

Yes i tried out the changes in a "real" setup, on our "preprod" stack, and made some tests for error Messages : all work fine :) 

This is "minor" updates for Archaius and Eureka Client, because Major versions need more refactoring to work properly (lots of changes on up-to-date versions !!!), also in this way there are less risks for integration problems.

I will change the POM version :)

Thanks,

Commit:
Change POM Version.

- Change POM version from 0.0.7-SNAPSHOT to 0.0.6-SNAPSHOT (the true next version).

Tetragramato commented:
OK, version changed :)

sawano commented:
Great I'll release a new version soon.

Tetragramato commented:
thanks :)
